compiler/vm.nim

```diff
@@ -718,10 +782,22 @@ proc rawExecute(c: PCtx, start: int, tos: PStackFrame): TFullReg =
         if regs[rb].node.kind == nkRefTy:
           regs[ra].node = regs[rb].node[0]
         else:
-          ensureKind(rkNode)
-          regs[ra].node = regs[rb].node
+          let node = regs[rb].node
+          let typ = node.typ
```

Comment:
Same.

Reply:
added a defensive `doAssert typ != nil` in the right place, but IIUC the tests i wrote would've caught it if it were nil